### PR TITLE
Allow using gist for crafting custom release announcement

### PR DESCRIPTION
**Pull request by premun**
Allows specifying a gist where the release announcement can be crafted and the release stage will use it later.
Not providing the gist URL will use the default template.
The announcement text can be previewed in the Pre-Release stage.

Resolves https://github.com/dotnet/source-build/issues/3270

Example gist: https://gist.github.com/premun/442ada4351185789bf462cd2b4075e78
Resulting announcement: https://dev.azure.com/dnceng/internal/_build/results?buildId=2128786&view=logs&j=19992227-62fb-5b50-4e29-3b72bd33eea1&t=36e702c7-c9ea-550a-b17d-5e4ea8b6e92e&l=78

### CODE REVIEW

File: eng/templates/stages/release.yml [hunk 1]
```diff
@@ -239,11 +242,23 @@ stages:
...
           fi
 
-          template="$(envsubst < source-build-release-announcement.md)"
-          # Get the line in the template that is prefixed with "Title:" and remove the prefix
-          title=$(echo "$template" | grep "^Title:" | cut -d " " -f2-)
-          # Get the inverse of the above selection
-          body=$(echo "$template" | grep -v "^Title:")
+          if [[ -z '${{ parameters.announcementGist }}' ]]; then
+            echo "Loading announcement template from source-build-release-announcement.md"
+            template="$(envsubst < source-build-release-announcement.md)"
```
Comment (premun) [inline eng/templates/stages/release.yml:247]: A bug, I need to use the new script here..

**Comment by premun** [comment]
@MichaelSimons this is tested and ready, I'd like to use it for 8.0 preview 2 release

File: .github/ISSUE_TEMPLATE/releases/release_checklist.md [hunk 2]
```diff
@@ -26,21 +26,10 @@
...
                 - ⚠️ 8.0: [dotnet-dotnet](https://dev.azure.com/dnceng/internal/_build?definitionId=1219) and the `dotnet/dotnet` commit that represents the release
+           - [ ] The `Create announcement draft` step should produce a valid announcement text. When using the announcement gist, copy the text to your gist file, copy the title to your gist's file name (add `.md`).
            - [ ] Ensure the `PoisonTests` and `SdkContentTests` are passing. Warnings indicate a baseline diff and should be inspected carefully.
                 - Please note that failures of these tests manifest as warnings in the `Run Tests` build step and not as failed tests in the test result viewer. This means you need to verify the `dotnet-dotnet` build doesn't have any warnings regarding these tests.
       1. - [ ] `Approval - Test prereqs` stage
-           - [ ] Gather smoke-test prereqs ([🔁 automation tracking issue](https://github.com/dotnet/source-build/issues/3068))
```
Comment (MichaelSimons) [inline .github/ISSUE_TEMPLATE/releases/release_checklist.md:32]: This shouldn't be removed this just yet.  We need to give our partners a period of time (minimum one servicing release) to switch from consuming the smoke-test-prereqs to use the artifacts feed.

File: eng/source-build-release-official.yml [hunk 3]
```diff
@@ -62,9 +62,13 @@ parameters:
   type: boolean
   default: true
 - name: createReleaseAnnouncement
-  displayName: Create Release Announcement
+  displayName: Create release announcement
```
Comment (MichaelSimons) [inline eng/source-build-release-official.yml:65]: FYI, we are rather inconsistent on the displayName casing across the board.  Most appear to use Title casing - Dyn Run, Submit Release PR, etc.

File: .github/ISSUE_TEMPLATE/releases/release_checklist.md (hunk 2 above)
Comment (premun) [reply .github/ISSUE_TEMPLATE/releases/release_checklist.md]: Makes sense

File: eng/create-announcement-draft.sh [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+{
+   echo "Uses a template to draft a release announcement and outputs it into a file."
+   echo
+   echo "Options:"
+   echo "--template            The base template to use for the announcement"
+   echo "--channel             Release channel, e.g. '8.0'"
+   echo "--release-name        Human readable name of the release, e.g. '.NET 8 Preview 1'"
+   echo "--sdk-version         The .NET SDK version that is being released"
+   echo "--runtime-version     The .NET runtime version that is being released"
+   echo "--tag                 The release tag, e.g. v8.0.0-preview.1"
```
Comment (crummel) [inline eng/create-announcement-draft.sh:14]: Might want to specify that the repository this is for varies by channel.

**Review by crummel — APPROVED** [review]
LGTM, one small nitpick on the script parameter help.

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**